### PR TITLE
Enable temporal scalability for h.264 from MSDK & external encoder.

### PR DESCRIPTION
**Pull request by taste1981**
Also added a new GlobalConfiguraiton API: SetH264TemporalLayers() to
specify layers requested. Default to 1 and temporal scalability
disabled.

### CODE REVIEW

File: talk/owt/sdk/include/cpp/owt/base/globalconfiguration.h [hunk 1]
```diff
@@ -83,6 +83,16 @@ class GlobalConfiguration {
           audio_frame_generator_.reset(nullptr);
       }
   }
+  /**
+   @brief This function sets the temporal layers for h264.
+   @param num_temporal_layers Number of temporal layers for h.264. Value greater
+   than 4 will be ignored.
+  */
+  static void SetH264EncoderTemporalLayers(int layers) {
```
Comment (jianjunz) [inline talk/owt/sdk/include/cpp/owt/base/globalconfiguration.h]: Temporal layer settings are expected to be configured through encoding parameters.

Comment (taste1981) [reply talk/owt/sdk/include/cpp/owt/base/globalconfiguration.h]: That's true. The code for configuring via Encoding parameters is also included in this patch. However, upstream does not really yet pass it to video engine, 
Also this is not in spec so setting via encoding parameter is not always guarenteed to be kept.

Comment (jianjunz) [reply talk/owt/sdk/include/cpp/owt/base/globalconfiguration.h]: It's defined in [Scalable Video Coding (SVC) Extension for WebRTC](https://www.w3.org/TR/webrtc-svc/), not webrtc-pc.

Please describe the reason in code, and remove this API when encoding parameters take effect in the future. They may conflict with each other.

Comment (taste1981) [reply talk/owt/sdk/include/cpp/owt/base/globalconfiguration.h]: done.

**Pull request merged**